tsdb/chunks/head_chunks.go

```diff
@@ -787,13 +918,19 @@ func (cdm *ChunkDiskMapper) Size() (int64, error) {
 	return fileutil.DirSize(cdm.dir.Name())
 }
 
-func (cdm *ChunkDiskMapper) curFileSize() int64 {
-	return cdm.curFileNumBytes.Load()
+func (cdm *ChunkDiskMapper) curFileSize() uint64 {
+	return cdm.curFileOffset.Load()
 }
 
 // Close closes all the open files in ChunkDiskMapper.
 // It is not longer safe to access chunks from this struct after calling Close.
 func (cdm *ChunkDiskMapper) Close() error {
+	// Locking the eventual position lock blocks WriteChunk()
+	cdm.evtlPosMtx.Lock()
+	defer cdm.evtlPosMtx.Unlock()
```

Comment:
Nitpick on naming: what if we call it `cmd.writeMtx` instead? Then it's clear than writeMtx blocks writes, plus since all the position assigning logic happens inside of the writes, it will be covered too.

Reply:
wouldn't that be confusing if there is already a `writePathMtx`?

the tricky thing about the chunk disk mapper is that it keeps track of chunks which are being written to the file and also chunks which will eventually be written to the file. I'm trying to separate these two things to not get them mixed up with each other because they need to have separate locks etc.

Reply:
Oh, I didn't notice that there's such mutex already, sorry.